### PR TITLE
Products fetched 

### PR DESCRIPTION
**Pull request by Monfernape**
Details:
Handled product fetching and display on UI when user click on specific category.

### CODE REVIEW

File: src/components/NavBar.js [hunk 1]
```diff
@@ -29,8 +30,30 @@ export function NavBar() {
     setOpen(false);
   };
 
-  const Cart = () => {
-    alert("You have added the product to your cart");
+  const products = useSelector((state) => state.productsReducer.Products);
+  console.log(products);
```
Comment (Monfernape) [inline src/components/NavBar.js:34]: remove the console

File: src/components/NavBar.js [hunk 2]
```diff
@@ -29,8 +30,30 @@ export function NavBar() {
...
+          .includes(searchValue.toUpperCase())
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+    SearchedProducts = newArray;
+    console.log(SearchedProducts);
```
Comment (Monfernape) [inline src/components/NavBar.js:56]: Plz remove the console

File: src/components/shopping redux/ServiceApi.js [hunk 3]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
       `https://backendapi.turing.com/categories
         `
     )
+    .then((response) => {
+      // console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:10]: Remove this comment

File: src/components/shopping redux/ServiceApi.js [hunk 4]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
...
+}
+
+export function GetProducts() {
+  return axios
+    .get(
+      `https://backendapi.turing.com/products/inCategory/1?limit=50
+        `
+    )
+    .then((response) => {
+      console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:22]: remove the console

File: src/components/shopping redux/ServiceApi.js [hunk 5]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
...
+
+export function GetProductsItalian() {
+  return axios
+    .get(
+      `https://backendapi.turing.com/products/inCategory/2?limit=50
+
+        `
+    )
     .then((response) => {
       console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:35]: remove the console

File: src/components/shopping redux/ServiceApi.js [hunk 6]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
...
+
+export function GetProductsIRISH() {
+  return axios
+    .get(
+      `https://backendapi.turing.com/products/inCategory/3?limit=50
+
+        `
+    )
+    .then((response) => {
+      console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:48]: remove the console

File: src/components/shopping redux/ServiceApi.js [hunk 7]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
...
+
+export function GetProductsAnimal() {
+  return axios
+    .get(
+      `https://backendapi.turing.com/products/inCategory/4?limit=50
+
+        `
+    )
+    .then((response) => {
+      console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:61]: console ????

File: src/components/shopping redux/ServiceApi.js [hunk 8]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
...
+
+export function GetProductsFlower() {
+  return axios
+    .get(
+      `https://backendapi.turing.com/products/inCategory/5?limit=50
+
+        `
+    )
+    .then((response) => {
+      console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:74]: console ???

File: src/components/shopping redux/ServiceApi.js [hunk 9]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
...
+
+export function GetProductsChristmas() {
+  return axios
+    .get(
+      `https://backendapi.turing.com/products/inCategory/6?limit=50
+
+        `
+    )
+    .then((response) => {
+      console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:87]: console

File: src/components/shopping redux/ServiceApi.js [hunk 10]
```diff
@@ -6,8 +6,100 @@ export function ServiceApi() {
...
+
+export function GetProductsValentines() {
+  return axios
+    .get(
+      `https://backendapi.turing.com/products/inCategory/7?limit=50
+
+        `
+    )
+    .then((response) => {
+      console.log(response.data.rows);
```
Comment (Monfernape) [inline src/components/shopping redux/ServiceApi.js:100]: console ???

File: src/components/shopping redux/store/Saga.js [hunk 11]
```diff
@@ -1,13 +1,98 @@
...
+  SET_PRODUCTS_LIST_CHRISTMAS,
+  GET_PRODUCTS_LIST_VALENTINES,
+  SET_PRODUCTS_LIST_VALENTINES,
+} from "../Actions";
 
-function* fetchCateogaries(action) {
-  const user = yield call(ServiceApi);
-  yield put({ type: SET_CATEOGARIES_LIST, payload: user });
+function* fetchProducts(action) {
+  console.log("action:", action);
```
Comment (Monfernape) [inline src/components/shopping redux/store/Saga.js:31]: console

File: src/components/shopping redux/store/Saga.js [hunk 12]
```diff
@@ -1,13 +1,98 @@
...
 }
 
-export function* fetchCateogariesSaga() {
-  yield takeEvery(GET_CATEOGARIES_LIST, fetchCateogaries);
+export function* fetchProductsSaga() {
+  yield takeEvery(GET_PRODUCTS_LIST, fetchProducts);
+}
+
+function* fetchProductsItalian(action) {
+  console.log("action:", action);
```
Comment (Monfernape) [inline src/components/shopping redux/store/Saga.js:41]: console

File: src/components/shopping redux/store/Saga.js [hunk 13]
```diff
@@ -1,13 +1,98 @@
...
+  const products = yield call(GetProductsIRISH);
+  yield put({ type: SET_PRODUCTS_LIST_IRISH, payload: products });
+}
+
+export function* fetchProductsIrishSaga() {
+  yield takeEvery(GET_PRODUCTS_LIST_IRISH, fetchProductsIrish);
+}
+
+function* fetchProductsAnimal(action) {
+  console.log("action:", action);
```
Comment (Monfernape) [inline src/components/shopping redux/store/Saga.js:61]: console ???

File: src/components/shopping redux/store/Saga.js [hunk 14]
```diff
@@ -1,13 +1,98 @@
...
+  const products = yield call(GetProductsAnimal);
+  yield put({ type: SET_PRODUCTS_LIST_ANIMAL, payload: products });
+}
+
+export function* fetchProductsAnimalSaga() {
+  yield takeEvery(GET_PRODUCTS_LIST_ANIMAL, fetchProductsAnimal);
+}
+
+function* fetchProductsFlower(action) {
+  console.log("action:", action);
```
Comment (Monfernape) [inline src/components/shopping redux/store/Saga.js:71]: console ????

File: src/components/shopping redux/store/Saga.js [hunk 15]
```diff
@@ -1,13 +1,98 @@
...
+  const products = yield call(GetProductsFlower);
+  yield put({ type: SET_PRODUCTS_LIST_FLOWER, payload: products });
+}
+
+export function* fetchProductsFlowerSaga() {
+  yield takeEvery(GET_PRODUCTS_LIST_FLOWER, fetchProductsFlower);
+}
+
+function* fetchProductsChristmas(action) {
+  console.log("action:", action);
```
Comment (Monfernape) [inline src/components/shopping redux/store/Saga.js:81]: console ???

File: src/components/shopping redux/store/Saga.js [hunk 16]
```diff
@@ -1,13 +1,98 @@
...
+  const products = yield call(GetProductsChristmas);
+  yield put({ type: SET_PRODUCTS_LIST_CHRISTMAS, payload: products });
+}
+
+export function* fetchProductsChristmasSaga() {
+  yield takeEvery(GET_PRODUCTS_LIST_CHRISTMAS, fetchProductsChristmas);
+}
+
+function* fetchProductsValentines(action) {
+  console.log("action:", action);
```
Comment (Monfernape) [inline src/components/shopping redux/store/Saga.js:91]: console ???

**Review by Monfernape — COMMENTED** [review]
remove all the consoles plzz

File: src/components/NavBar.js (hunk 1 above)
Comment (usmangq12) [reply src/components/NavBar.js:34]: Console Removed

File: src/components/shopping redux/ServiceApi.js (hunk 3 above)
Comment (usmangq12) [reply src/components/shopping redux/ServiceApi.js:10]: Okay

**Pull request merged**